tests/subsys/net/lib/fota_download/src/test_parse_dual_resource_locator.c

```diff
@@ -0,0 +1,172 @@
+ * @param expected_selection - The resource locator that parse_dual_resource_locator is expected to
+ *			       select
+ * @param s0_active - The slot activity state to be fed to parse_dual_resource_locator
+ */
+void test_fota_download_parse_dual_resource_locator_generic(const char * const resource_locator,
+						    const char * const expected_selection,
+						    const bool s0_active)
+{
+	int err;
+	const char *update;
+
+	strcpy(buf, resource_locator);
+	err = fota_download_parse_dual_resource_locator(buf, s0_active, &update);
+
+	zassert_equal(err, 0, "Selection of resource locator from valid dual resource locator "
+			      " should not fail.");
+	if (expected_selection == NULL) {
+		zassert_is_null(update, "fota_download_parse_dual_resource_locator was expected to "
+					"select NULL, and did not.");
+	} else {
+		zassert_not_null(update, "fota_download_parse_dual_resource_locator did not select "
+					 "a resource locator");
+		zassert_true(strcmp(expected_selection, update) == 0,
+		     "fota_download_parse_dual_resource_locator did not select the expected "
+		     "resource locator.");
+	}
+
+}
+
+ZTEST(fota_download_parse_dual_resource_locator_tests, test_dual_A_s0_active)
+{
+	test_fota_download_parse_dual_resource_locator_generic(S0_A " " S1_A, S1_A, S0_ACTIVE);
+}
+
+ZTEST(fota_download_parse_dual_resource_locator_tests, test_dual_A_s1_active)
+{
+	test_fota_download_parse_dual_resource_locator_generic(S0_A " " S1_A, S0_A, S1_ACTIVE);
+}
+
+ZTEST(fota_download_parse_dual_resource_locator_tests, test_dual_B_s0_active)
+{
+	test_fota_download_parse_dual_resource_locator_generic(S0_B " " S1_B, S1_B, S0_ACTIVE);
+}
+
+ZTEST(fota_download_parse_dual_resource_locator_tests, test_dual_B_s1_active)
+{
+	test_fota_download_parse_dual_resource_locator_generic(S0_B " " S1_B, S0_B, S1_ACTIVE);
+}
+
+ZTEST(fota_download_parse_dual_resource_locator_tests, test_dual_C_s0_active)
+{
+	test_fota_download_parse_dual_resource_locator_generic(S0_C " " S1_C, S1_C, S0_ACTIVE);
+}
+
+ZTEST(fota_download_parse_dual_resource_locator_tests, test_dual_C_s1_active)
+{
+	test_fota_download_parse_dual_resource_locator_generic(S0_C " " S1_C, S0_C, S1_ACTIVE);
+}
+
+ZTEST(fota_download_parse_dual_resource_locator_tests, test_single_A)
+{
+	test_fota_download_parse_dual_resource_locator_generic(S0_A, NULL, S1_ACTIVE);
+}
+
+ZTEST(fota_download_parse_dual_resource_locator_tests, test_single_B)
+{
+	test_fota_download_parse_dual_resource_locator_generic(S1_B, NULL, S1_ACTIVE);
+}
+
+ZTEST(fota_download_parse_dual_resource_locator_tests, test_single_C)
+{
+	test_fota_download_parse_dual_resource_locator_generic(S0_C, NULL, S0_ACTIVE);
+}
+
+#ifdef CONFIG_BOARD_NATIVE_POSIX
+ZTEST(fota_download_parse_dual_resource_locator_tests, test_ptr_placement)
+{
+	/* This test cannot be executed on native posix as there is
+	 * no mechanism in place do decude if an address maps to non-volatile
+	 * storage or RAM
+	 */
+	ztest_test_skip();
```

Comment:
Friendly suggestion: ZTEST new API has a shorter way of skipping tests, you can use: `Z_TEST_SKIP_IFNDEF(CONFIG_[something]);` or ``Z_TEST_SKIP_IFDEF(CONFIG_[something]);` and do not have to make another copy of the same test function just to skip it.

Reply:
Neat, I will adopt this.

In fact, we have flat out dropped `NATIVE_POSIX` support from these tests, so the entire check could be removed. But I think it is harmless to allow such a simple skip line to remain in the code-base in case `NATIVE_POSIX` is ever re-added.